.github/workflows/deploy_to_do.yml

```
@@ -0,0 +1,155 @@
+          oauth-secret: ${{ secrets.TS_OAUTH_SECRET }} # Tailscale OAuth secret
+          tags: tag:github-actions # Tag for this Tailscale node
+          version: latest # Use latest Tailscale version
+          use-cache: 'true' # Cache Tailscale for faster setup
+
+      - name: Debug ED25519 Key # Setup and verify SSH key
+        run: |
+          mkdir -p ~/.ssh # Create SSH directory
+          echo "${{ secrets.DO_SSH_KEY }}" > ~/.ssh/id_ed25519 # Save SSH key
+          chmod 600 ~/.ssh/id_ed25519 # Set correct permissions
+          
+          echo "${{ secrets.DO_HOST_KEY }}" >> ~/.ssh/known_hosts # Add host key
+          
+          echo "Key information:" # Display key information for debugging
+          ssh-keygen -l -f ~/.ssh/id_ed25519 # Show key fingerprint
+          
+          echo "Public key that should be in authorized_keys:" # Show public key
+          ssh-keygen -y -f ~/.ssh/id_ed25519 # Generate public key from private
+          
+          echo "Attempting connection with verbose output:" # Test connection
+          ssh -vvv -o StrictHostKeyChecking=accept-new -o BatchMode=yes -i ~/.ssh/id_ed25519 ${{ secrets.DO_USERNAME }}@${{ secrets.DO_TAILSCALE_NAME }} 'echo "Connection successful"'
+
+      # ✅ Step 2: SSH and deploy to the droplet over Tailscale IP
+      - name: SSH and Deploy with Docker Compose # Main deployment step
+        run: |
+          mkdir -p ~/.ssh # Create SSH directory
+          echo "${{ secrets.DO_SSH_KEY }}" > ~/.ssh/id_ed25519 # Save SSH key
+          chmod 600 ~/.ssh/id_ed25519 # Set correct permissions
+          
+          ssh-keyscan -H ${{ secrets.DO_TAILSCALE_NAME }} >> ~/.ssh/known_hosts # Add host key
+
+          # Deploy with SSH - add multiple options to disable all prompts
+          ssh -o StrictHostKeyChecking=accept-new -o BatchMode=yes -i ~/.ssh/id_ed25519 ${{ secrets.DO_USERNAME }}@${{ secrets.DO_TAILSCALE_NAME }} '
+            set -e # Exit on any error
+
+            echo "📦 Starting manual deployment to Tailscale-connected droplet..." # Announce start
+            cd /home/deploy # Change to deployment directory
+            
+            echo "📥 Retrieving the docker-compose.yaml" # Get compose file first
+            curl -O https://raw.githubusercontent.com/refactor-group/refactor-platform-rs/refs/heads/main/docker-compose.yaml # Download compose
+            
+            echo "📝 Creating .env file with GitHub secrets..." # Create env file
+            cat > .env << 'EOF'
+            # ==============================
+            #   PostgreSQL Configuration
+            # ==============================
+            POSTGRES_USER=${{ secrets.POSTGRES_USER }}
+            POSTGRES_PASSWORD=${{ secrets.POSTGRES_PASSWORD }}
+            POSTGRES_DB=${{ secrets.POSTGRES_DB }}
+            POSTGRES_HOST=postgres
+            POSTGRES_PORT=${{ secrets.POSTGRES_PORT }}
+            POSTGRES_SCHEMA=${{ secrets.POSTGRES_SCHEMA }}
+
```

```suggestion
            # ==============================
            #   Database URL (computed)
            # ==============================
            DATABASE_URL=postgres://${{ secrets.POSTGRES_USER }}:${{ secrets.POSTGRES_PASSWORD }}@postgres:${{ secrets.POSTGRES_PORT }}/${{ secrets.POSTGRES_DB }}

```

Moving this to group it with like-environment variables.